__tests__/articles.test.js

```diff
@@ -0,0 +1,53 @@
+const request = require("supertest");
+const db = require("../db/connection");
+const seed = require("../db/seeds/seed");
+const app = require("../app");
+const {
+  topicData,
+  userData,
+  articleData,
+  commentData,
+} = require("../db/data/test-data/index");
+
+beforeEach(() => {
+  return seed({ topicData, userData, articleData, commentData });
+});
+
+afterAll(() => {
+  return db.end();
+});
+
+describe("/api/articles/:article_id", () => {
+  test("200 - GET - will return an article object with the given article_id, with the properties of author, title, article_id, body, topic, created_at, votes and article_img_url", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then((res) => {
+        const article = res.body.article;
+        expect(article).toHaveProperty("article_id", expect.any(Number));
+        expect(article).toHaveProperty("title", expect.any(String));
+        expect(article).toHaveProperty("topic", expect.any(String));
+        expect(article).toHaveProperty("author", expect.any(String));
+        expect(article).toHaveProperty("body", expect.any(String));
+        expect(article).toHaveProperty("created_at", expect.any(String));
+        expect(article).toHaveProperty("votes", expect.any(Number));
+        expect(article).toHaveProperty("article_img_url", expect.any(String));
+      });
+  });
+  test("404 - GET - sends an appropriate status and error message when a non-exist article_id is used", () => {
+    return request(app)
+      .get("/api/articles/1000000000000000")
```

Comment:
Be careful testing with numbers this big, you could get a different error all together about the search being out of range. A simple 99 would suffice.

Reply:
<img width="988" alt="Screenshot 2024-08-26 at 13 14 42" src="https://github.com/user-attachments/assets/b5378fa1-2146-4405-baf3-70bac36d40fc">